apps/cowswap-frontend/src/modules/combinedBalances/updater/BalancesCombinedUpdater.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect } from 'react'
+
+import { BalancesState, useTokensBalances } from '@cowprotocol/balances-and-allowances'
+import { useWalletInfo } from '@cowprotocol/wallet'
+
+import { BigNumber } from 'ethers'
+
+import { useHooks } from 'modules/hooksStore'
+import { usePreHookBalanceDiff } from 'modules/hooksStore/hooks/useBalancesDiff'
+import { useIsHooksTradeType } from 'modules/trade'
+import { useSetAtom } from 'jotai'
+import { balancesCombinedAtom } from '../state/balanceCombinedAtom'
+
+export function BalancesCombinedUpdater() {
+  const { account } = useWalletInfo()
+  const setBalancesCombined = useSetAtom(balancesCombinedAtom)
+  const preHooksBalancesDiff = usePreHookBalanceDiff()
+  const { preHooks } = useHooks()
+  const tokenBalances = useTokensBalances()
+  const isHooksTradeType = useIsHooksTradeType()
+
+  useEffect(() => {
+    if (!account || !isHooksTradeType || !preHooks.length) {
+      setBalancesCombined(tokenBalances)
+      return
+    }
+    const accountBalancesDiff = preHooksBalancesDiff[account.toLowerCase()] || {}
+    setBalancesCombined(applyBalanceDiffs(tokenBalances, accountBalancesDiff))
```

Comment:
Do you know why this is being re-computed still every few seconds? 

<img width="1402" alt="image" src="https://github.com/user-attachments/assets/d1687e30-874a-4cdf-9170-14aba1c866ec">

Maybe this can help to figure it out https://github.com/simbathesailor/use-what-changed

Reply:
IIUC is because the `tokenBalances` state is updated every few seconds.